src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -580,22 +577,10 @@ private Set<ProgrammingExerciseTestCase> filterTestCasesForStudents(final Set<Pr
      */
     private Result calculateScoreForResult(Set<ProgrammingExerciseTestCase> testCases, Set<ProgrammingExerciseTestCase> testCasesForCurrentDate, @NotNull Result result,
             ProgrammingExercise exercise, boolean applySubmissionPolicy) {
-        List<Feedback> testCaseFeedback = new ArrayList<>();
-        List<Feedback> staticCodeAnalysisFeedback = new ArrayList<>();
-        for (var feedback : result.getFeedbacks()) {
-            if (feedback.getType() != FeedbackType.AUTOMATIC) {
-                continue;
-            }
+        List<Feedback> staticCodeAnalysisFeedback = result.getFeedbacks().stream().filter(Feedback::isStaticCodeAnalysisFeedback).toList();
+        List<Feedback> testCaseFeedback = result.getFeedbacks().stream().filter(Feedback::isTestFeedback).toList();
 
-            if (feedback.isStaticCodeAnalysisFeedback()) {
-                staticCodeAnalysisFeedback.add(feedback);
-            }
-            else {
-                testCaseFeedback.add(feedback);
-            }
-        }
-
-        // Remove feedback that is in an invisible sca category
+        // Remove feedback that is in an invisible SCA category
```

Comment:
The previous version has the advantage that we only need to loop once over the feedbacks to group them into categories. Now it’s twice.